### ISSUE TITLE
type check needed in io.fits.hdu.image

### ISSUE THREAD

**Issue by cgosmeyer** [astropy/astropy#5136]
A colleague and I had trouble opening a FITS file with astropy (though it worked fine in IDL)

```
> from astropy.io.fits import getdata
> h = getdata('gj1214_intx1d.fits')
```
Resulting in the error:
```
/usr/stsci/ssbdev/python/lib/python2.7/site-packages/astropy/io/fits/hdu/image.pyc in __init__(self, data, header, do_not_scale_image_data, uint, scale_back, ignore_blank, **kwargs)
    106         # the image data
    107         self._axes = [self._header.get('NAXIS' + str(axis + 1), 0)
--> 108                       for axis in range(self._header.get('NAXIS', 0))]
    109         self._bitpix = self._header.get('BITPIX', 8)
    110         self._gcount = self._header.get('GCOUNT', 1)
TypeError: range() integer end argument expected, got str.
```

The issue appeared to be, in this file's case, that the BITPIX and NAXIS values were strings. When we changed the values to integers, astropy accepted the file.   So is it possible to add a check that converts these values to integers in these corner cases?

**Comment by pllim** [comment]
Does your data pass the FITS verification? If not, then you should fix the data.

**Comment by astrofrog** [comment]
@pllim - I think @cgosmeyer meant that they know the file is invalid, and is asking if we can add a workaround. My inclination is no - this is clearly an invalid file. @cgosmeyer - where did the file come from?

**Comment by pllim** [comment]
@astrofrog , okay... I do agree that we should not support invalid formats. To support one open a precedence to a can of worms.

**Comment by astrofrog** [comment]
I think the only situation where we should consider bending that rule (and we have in the past) is if say all official 2MASS data or HST data had an issue, and it was impossible to expect the files to be made again. But if this is just a one-off file, then I think it's best to just fix it at the source.

**Comment by cgosmeyer** [comment]
Yes, the FITS file was definitely in error and we did fix the two keyword values in the end.  I think my gripe is that I could not even open the header using astropy to see what the problem was -- I resorted to IDL.  I do see what you mean that you don't want to start down the slope of checking FITS headers, though.

**Comment by pllim** [comment]
@cgosmeyer , did you try using [fitscheck](https://github.com/astropy/astropy/blob/master/astropy/io/fits/scripts/fitscheck.py) ?

**Comment by astrofrog** [comment]
For an independent check, you can also use http://fits.gsfc.nasa.gov/fits_verify.html

**Comment by cgosmeyer** [comment]
Thanks!  I was not aware that astropy had a fits check!

**Comment by embray** [comment]
Regardless this should be handled better by verifying that the header is correct before trying to read the data and blowing up in some non-obvious way.
